packages/skygear-core/lib/cloud/registry.js

```diff
@@ -143,7 +143,32 @@ export class Registry {
   }
 
   getHandler(name, method) {
-    return this.handlers[name][method];
+    for (let k in this.handlers) {
+      if (this._matchHandler(k, name))
+        return this.handlers[k][method];
+    }
+    return undefined;
+  }
+
+  parseParamsInPath(path) {
```

Comment:
The registry responsibility is not parsing the incoming request.

`parseParamsInPath` belong to `SkygearRequest` should make more semantic sense. Take a look at current implementation of `SkygearRequest` constructor, it already contains the logic of parsing incoming request.